extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/input/SlicerUtils.java

```diff
@@ -35,7 +35,7 @@
    *
    * Items are assigned round-robin.
    */
-  public static <T> List<List<T>> makeSlices(final Iterator<T> iterator, final int numSlices)
+  public static <T> List<List<T>> makeSlicesStatic(final Iterator<T> iterator, final int numSlices)
```

Comment:
Dropping this comment here since there is no good location otherwise: we seem to be computing slices in the controller. Does the controller adjust its task ask based on input size? If so, isn't there a negotiation?

* Check the inputs and compute total size & ideal task count.
* Adjust based on some configured absolute maximum task count.
* Ask Overlord for that number of workers. Get back an available task count (for non-auto-scale systems.)
* Revise the work allocation to bin-pack the actual available workers.

Or, is the allocation more static than this, based on configured values somewhere?

Reply:
There's no negotiation at this time; the controller comes up with a number of tasks it wants (up to the user-provided `maxNumTasks` setting) and then launches those tasks. A negotiation would be neat but would require more Overlord support than is currently possible.